catalog/spatial/registry/registry-federation-admin-impl/src/main/java/org/codice/ddf/registry/federationadmin/impl/FederationAdmin.java

```diff
@@ -0,0 +1,699 @@
+
+        Metacard metacard = getRegistryMetacardFromRegistryPackage(registryPackage);
+        metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE,
+                true));
+        return federationAdminService.addRegistryEntry(metacard);
+    }
+
+    @Override
+    public String createLocalEntry(String base64EncodedXmlData) throws FederationAdminException {
+        if (StringUtils.isBlank(base64EncodedXmlData)) {
+            throw new FederationAdminException(
+                    "Error creating local entry. String provided was blank.");
+        }
+
+        String metacardId;
+        try (InputStream xmlStream = new ByteArrayInputStream(Base64.getDecoder()
+                .decode(base64EncodedXmlData))) {
+            Metacard metacard = getRegistryMetacardFromInputStream(xmlStream);
+            metacard.setAttribute(new AttributeImpl(RegistryObjectMetacardType.REGISTRY_LOCAL_NODE,
+                    true));
+            metacardId = federationAdminService.addRegistryEntry(metacard);
+        } catch (IOException | IllegalArgumentException e) {
+            String message = "Error creating local entry. Couldn't decode string.";
+            LOGGER.error("{} Base64 encoded xml: {}", message, base64EncodedXmlData);
+            throw new FederationAdminException(message, e);
+        }
+        return metacardId;
+    }
+
+    @Override
+    public void updateLocalEntry(Map<String, Object> registryObjectMap)
+            throws FederationAdminException {
+
+        Optional<RegistryPackageType> registryPackageOptional = registryTypeConverter.convert(
+                registryObjectMap);
+        RegistryPackageType registryPackage =
+                registryPackageOptional.orElseThrow(() -> new FederationAdminException(
+                        "Error updating local registry entry. Couldn't convert registry map to a registry package."));
+
+        updateDateFields(registryPackage);
+
+        List<Metacard> existingMetacards =
+                federationAdminService.getLocalRegistryMetacardsByRegistryIds(Collections.singletonList(
+                        registryPackage.getId()));
+
+        if (CollectionUtils.isEmpty(existingMetacards)) {
+            String message = "Error updating local registry entry. Registry metacard not found.";
+            LOGGER.error("{} Registry ID: {}", message, registryPackage.getId());
+            throw new FederationAdminException(message);
+        }
+
+        if (existingMetacards.size() > 1) {
+            String message =
+                    "Error updating local registry entry. Multiple registry metacards found.";
+
+            List<String> metacardIds = new ArrayList<>();
+            metacardIds.addAll(existingMetacards.stream()
+                    .map(Metacard::getId)
+                    .collect(Collectors.toList()));
+            LOGGER.error("{} Matching registry metacard ids: {}", message, metacardIds);
+
```

Comment:
Probably not. Removed logging statement and just threw the exception with the basic message, no ID's